### ISSUE TITLE
Build failure on macOS 11.3 (case-insensitive) due to VERSION file included as C++20 <version> header

### ISSUE THREAD

**Issue by cho-m** [mobile-shell/mosh#1140]
For reference, this issue was seen in Homebrew while updating `protobuf`: https://github.com/Homebrew/homebrew-core/pull/77373#issuecomment-841783566

Also, my attempt at quick (hack-ish) patch of the generated `Makefile.in`, so Homebrew isn't blocked: https://github.com/Homebrew/homebrew-core/pull/77373#issuecomment-841942117

The exact build environment that causes issue is:
- macOS Big Sur 11.3+ (XCode 12.5), which has new C++20 `<version>` header
- Default case-insensitive filesystem on macOS.
- Tried on last release `mosh 1.3.2`

Example `make` failure:
```make
❯ ./configure
checking for a BSD-compatible install... /opt/homebrew/bin/ginstall -c
checking whether build environment is sane... yes
...

❯ make
/Applications/Xcode.app/Contents/Developer/usr/bin/make  all-recursive
Making all in scripts
Making all in src
Making all in protobufs
  GEN      userinput.pb.cc
  GEN      hostinput.pb.cc
  GEN      transportinstruction.pb.cc
/Applications/Xcode.app/Contents/Developer/usr/bin/make  all-am
  CXX      userinput.pb.o
In file included from userinput.pb.cc:4:
In file included from ./userinput.pb.h:7:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/limits:105:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/type_traits:417:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/cstddef:37:
../../version:1:1: error: unknown type name 'mosh'
mosh 1.3.2
^
../../version:1:6: error: expected unqualified-id
mosh 1.3.2
     ^

...
```

Due to case-insensitive filesystem, macOS sees `VERSION` file equal to `version` file.
As directory is in include path, it is assumed to be the `<version>` header, which causes wrong C++ include.

Is there a recommended way to deal with this compared to quick rename patch that I attempted?

**Comment by eminence** [comment]
What version/commit of mosh are you building?  Can you try the latest commit from the master branch?

This sounds a lot like the issue reported in #1051 and fixed in #1052

**Comment by cho-m** [comment]
For Homebrew, we need to build stable release, so the code is the older **1.3.2** release.

At quick glance, those issues do look related. Will take a look and see if we can apply patch.

**Comment by cho-m** [comment]
@eminence It should be same issue, so this can probably be closed. 

I didn't notice that previous issue as I wasn't aware of problem being identified back in 2019.
We are currently seeing this in Homebrew as Apple has updated the XCode recently with macOS 11.3.

For Homebrew, we will probably try a variant of the fix for **1.3.2**.
There is too much of a gap in code to use diff as is..
